wp-modules/app/js/src/components/PatternPreview/test/PreviewIframe.spec.tsx

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable import/default, camelcase */
+
+import { create } from 'react-test-renderer';
+import PreviewIframe from '../PreviewIframe';
+import WP_Element from '@wordpress/element';
+
+const FAKE_URL = 'http://some-fake-url.com';
+const VIEWPORT_WIDTH = 1200;
+const SCALE = 0.5;
+const STATE_MOCK_$iframeRef = {
+	contentWindow: {
+		document: {
+			documentElement: {
+				scrollHeight: 1200,
+			},
+		},
+	},
+};
+const STATE_MOCK_$iframeBodyHeight = 600;
+const STATE_MOCK_$iframeLoaded = true;
+
+jest.mock( '@wordpress/element', () => {
+	return {
+		...jest.requireActual( '@wordpress/element' ),
+		useState: jest.fn(),
+	};
+} );
+
+describe( 'PreviewIframe', () => {
+	beforeEach( () => {
+		jest.spyOn( WP_Element, 'useState' )
+			.mockImplementationOnce( () => [
+				STATE_MOCK_$iframeRef,
+				() => null,
+			] )
+			.mockImplementationOnce( () => [
+				STATE_MOCK_$iframeBodyHeight,
+				() => null,
+			] )
+			.mockImplementationOnce( () => [
+				STATE_MOCK_$iframeLoaded,
+				() => null,
+			] );
+	} );
+
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	it( 'renders an container div with the expected height and opacity style props', () => {
+		const testRenderer = create(
+			<PreviewIframe
+				url={ FAKE_URL }
+				scale={ SCALE }
+				viewportWidth={ VIEWPORT_WIDTH }
+			/>
+		);
+
+		expect(
+			testRenderer.root.findByProps( {
+				'data-testid': 'pattern-preview-iframe-inner',
+			} ).props
+		).toMatchObject( {
+			style: {
+				height: STATE_MOCK_$iframeBodyHeight * SCALE,
+				opacity: 1, // Opacity of `1` would otherwise be assigned by onLoad().
+			},
+		} );
+
+		testRenderer.unmount();
+	} );
+
+	it( 'renders an iframe with the expected props', () => {
+		const testRenderer = create(
+			<PreviewIframe
+				url={ FAKE_URL }
+				scale={ SCALE }
+				viewportWidth={ VIEWPORT_WIDTH }
+			/>
+		);
+
+		expect( testRenderer.root.findByType( 'iframe' ).props ).toMatchObject(
+			{
+				src: FAKE_URL,
+				style: {
+					width: VIEWPORT_WIDTH,
+					height: 600,
+				},
+			}
+		);
+
+		testRenderer.unmount();
+	} );
+
+	it( 'hides the `pattern-loader` div when loaded', () => {
```

Comment:
Very true! The setup is really complicated for these tests. I am definitely not opposed to simplifying it.

I do like the idea of testing some of the inner logic (such as the calculated height), though I admit it might not offer much benefit with this stubbed implementation.

Since I want to revisit this component to see if I can get rid of some state anyway, the testing for that iteration would have far less setup 😎

Reply:
I went ahead and pasted in this suggestion verbatim with [829efd4](https://github.com/studiopress/pattern-manager/pull/52/commits/829efd41ed4bbe40073fbe9ed3b057455ed13145).

Let's keep it simple and revisit as needed!

I was not a big fan of the hack I used for stubbing those state values anyway — I suspect it would've ultimately proven to be too brittle.